mobile/src/main/java/org/openhab/habdroid/model/Widget.kt

```diff
@@ -373,6 +401,51 @@ fun JSONObject.collectWidgets(parent: Widget?): List<Widget> {
     val icon = optStringOrNull("icon")
     val staticIcon = optBoolean("staticIcon", false)
 
+    val widgets = if (type == Widget.Type.Buttongrid) {
+        val buttonWidgets = ArrayList<Widget>(mappings.map { mapping ->
```

Comment:
Does `map` not already yield a `List`?

Reply:
Yes, but apparently it's an immutable one, so the subsequent attempt to add to the list results in an error.

Reply:
The more Kotlin-esque way to fix that is `.map { ... }.toMutableList()